### PR TITLE
Heat exchange models

### PR DESCRIPTION
**Pull request by spahrenk**
## Description
- In this PR, the form `Model` refers to a manufacturer product; `model` refers to a physics representation.
- This code simulates heat-exchange HPWH Models (AquaThermAire), in which tank fluid (currently water) remains isolated from a delivered-water line. A simple heat-exchange model was implemented for initial testing. Assigning the flag hasHeatExchanger as `true` applies this model. The model assumes heat transfer from each tank node to the inlet water line using an effectiveness coefficient (heatExchangerEffectiveness). This process is repeated for each node upward, in sequence. However, the initial AquaThermAire Model preset and file use a single node, so this model has not been directly tested.
- One additional test schedule (villara_24hr67) has been added. This is run for the AquaThermAire Model preset and file. No regression test is performed for this Model, lacking an accepted output file for comparison. Test profile was changed to convert degF to degC and DR flag was changed to unlock condenser. Passes all HPWH and cse tests.

## Author Progress Checklist:

- [X] Open draft pull request
    - [X] Make title clearly understandable in a standalone change log context
    - [X] Assign yourself the issue
    - [X] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [X] Make code changes (if you haven't already)
- [X] Self-review of code
    - [X] My code follows the style guidelines of this project
    - [X] I have added comments to my code, particularly in hard-to-understand areas
    - [X] I have only committed the necessary changes for this fix or feature
    - [X] I have made corresponding changes to the documentation
    - [X] My changes generate no new warnings
    - [X] I have ensured that my fix is effective or that my feature works as intended by:
        - [x] exercising the code changes in the test framework, and/or
        - [X] manually verifying the changes (as explained in the the pull request description above)
    - [X] My changes pass all local tests
    - [X] My changes successfully passes CI checks
    - [X] Add any unit test for proof and documentation.
    - [X] Merge in main branch and address resulting conflicts and/or test failures.
- [X] Move pull request out of draft mode and assign reviewers
- [] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

File: src/HPWHpresets.cc [hunk 1]
```diff
@@ -515,7 +515,7 @@ int HPWH::HPWHinit_presets(MODELS presetNum) {
 
 	else if (presetNum == MODELS_StorageTank) {
 		setNumNodes(12);
-		setpoint_C = F_TO_C(127.0);
+		setpoint_C = 800.;
```
Comment (nealkruis) [inline src/HPWHpresets.cc]: This shouldn't be changing on this branch, and will likely cause conflicts when master is merged in.

File: test/AquaThermAire.txt [hunk 2]
```diff
@@ -0,0 +1,71 @@
+verbosity silent
+numNodes 1 #number of nodes
```
Comment (nealkruis) [inline test/AquaThermAire.txt]: Change to 12?

File: test/AquaThermAire.txt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+setpoint 125 F
+volume 54.4 gal
+UA 10.35 kJperHrC
+depressTemp false
+mixOnDraw true
```
Comment (nealkruis) [inline test/AquaThermAire.txt]: This should be false, the draw doesn't go into the tank at all, and it shouldn't cause any mixing of the water.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: test/AquaThermAire.txt (hunk 2 above)
Comment (spahrenk) [reply test/AquaThermAire.txt]: done

File: test/AquaThermAire.txt (hunk 3 above)
Comment (spahrenk) [reply test/AquaThermAire.txt]: done

**Comment by spahrenk** [comment]
Added function bottomTwelfth_absolute. Fixed function topThird_absolute (seems to be unused.)

**Comment by spahrenk** [comment]
This is a primitive solution using the existing method to read hpwh spec files. Can improve with generalized-logic-node method (forthcoming).

**Comment by spahrenk** [comment]
Please wait... fails cse tests.

File: src/HPWH.cc [hunk 4]
```diff
@@ -63,6 +63,10 @@ const double HPWH::MAXOUTLET_R410A = F_TO_C(140.);
 const double HPWH::MAXOUTLET_R744 = F_TO_C(190.);
 const double HPWH::MINSINGLEPASSLIFT = dF_TO_dC(15.);
 
+double makeC(const double T_F_or_C,const HPWH::UNITS units,const bool absolute){
```
Comment (nealkruis) [inline src/HPWH.cc]: This can probably be inline, no?

The name is also not very descriptive. Maybe `convertTempToC`?

File: src/HPWH.cc [hunk 5]
```diff
@@ -1798,6 +1810,11 @@ std::shared_ptr<HPWH::TempBasedHeatingLogic> HPWH::bottomTwelfth(double decision
 	return std::make_shared<HPWH::TempBasedHeatingLogic>("bottom twelfth",nodeWeights,decisionPoint,this);
 }
 
+std::shared_ptr<HPWH::TempBasedHeatingLogic> HPWH::bottomTwelfth_absolute(double decisionPoint) {
+	std::vector<NodeWeight> nodeWeights = getNodeWeightRange(0., 1./12.);
+	return std::make_shared<HPWH::TempBasedHeatingLogic>("bottom twelfth",nodeWeights,decisionPoint,this,true);
+}
```
Comment (nealkruis) [inline src/HPWH.cc]: I don't think this is needed anymore.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: test/AquaThermAire.txt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+heatsource 0 inPowT4quad 0.0
+heatsource 0 copT4const -0.47273
+heatsource 0 copT4lin 0.042082
+heatsource 0 copT4quad 0.0
+
+
+heatsource 0 minT -25 F
+heatsource 0 maxT 125 F
+heatsource 0 hysteresis 1 F
+heatsource 0 coilConfig wrapped
```
Comment (nealkruis) [inline test/AquaThermAire.txt]: Should probably be "submerged" based on conversations with Ben. What impact will this have on the simulation?

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: test/AquaThermAire.txt (hunk 6 above)
Comment (spahrenk) [reply test/AquaThermAire.txt]: Wrapped and submerged use different heat distributions. Simulation results are slightly altered.

**Review by nealkruis — APPROVED** [review]

**Pull request merged**